### PR TITLE
Add `scale_quality` parameter to Texture.__init__

### PR DESCRIPTION
**Pull request by yunline**
Example:
```python
import pygame
from pygame._sdl2 import video

win=video.Window()
r=video.Renderer(win)

sf=pygame.image.load("cobweb.png")

tex_nearest=video.Texture(r,(16,16),scale_quality=video.SCALEQUALITY_NEAREST)
tex_nearest.update(sf)
tex_linear=video.Texture(r,(16,16),scale_quality=video.SCALEQUALITY_LINEAR)
tex_linear.update(sf)
tex_best=video.Texture(r,(16,16),scale_quality=video.SCALEQUALITY_BEST)
tex_best.update(sf)

tex_nearest.draw(dstrect=(10,10,16,16)) # original

tex_nearest.draw(dstrect=(10,128,128,128)) # nearest
tex_linear.draw(dstrect=(148,128,128,128)) # linear
tex_best.draw(dstrect=(286,128,128,128)) # best (Direct3D only)
r.present()

while 1:
    pygame.event.get()
```
image

### CODE REVIEW

**Comment by yunline** [comment]
I'm not sure if the ‘beset’ mode will crash on platforms without Direct3D

**Review by MyreMylar — APPROVED** [review]
LGTM 👍

**Comment by MyreMylar** [comment]
Another merge conflict. with video.c I guess this is cython issues rearing their heads again.

File: src_c/cython/pygame/_sdl2/video.pyx [hunk 1]
```diff
@@ -609,6 +613,13 @@ cdef class Texture:
                                       width, height)
         if not self._tex:
             raise error()
+            
+        if not scale_quality is None:
+            if SDL_VERSION_ATLEAST(2,0,12):
+                SDL_SetTextureScaleMode(self._tex,scale_quality)
+            else:
+                raise error("scale_quality requires SDL 2.0.12 or newer")
```
Comment (pmp-p) [inline src_c/cython/pygame/_sdl2/video.pyx:621]: does it have to be an error, old could it be a warning and fallback to some default scaling on older sdl ?

**Review by pmp-p — APPROVED** [review]

**Review by pmp-p — APPROVED** [review]

**Pull request merged**